rest-service/manager_rest/test/endpoints/test_plugins_updates.py

```diff
@@ -0,0 +1,365 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import uuid
+from copy import deepcopy
+
+import unittest
+from mock import patch
+from functools import wraps
+
+from manager_rest.test import base_test
+from manager_rest.test.attribute import attr
+from manager_rest.test.base_test import BaseServerTestCase
+
+from manager_rest.plugin_update.constants import STATES
+from manager_rest.manager_exceptions import NotFoundError
+from manager_rest.constants import HIDDEN_BLUEPRINT_SUFFIX
+from manager_rest.storage import get_storage_manager, models
+from manager_rest.plugin_update.manager import (PluginsUpdateManager,
+                                                get_plugins_updates_manager,
+                                                _did_plugins_to_install_change)
+
+from cloudify_rest_client.exceptions import CloudifyClientError
+
+from cloudify.models_states import ExecutionState
+
+from dsl_parser.constants import (PLUGIN_NAME_KEY,
+                                  WORKFLOW_PLUGINS_TO_INSTALL,
+                                  DEPLOYMENT_PLUGINS_TO_INSTALL)
+
+
+def suppress_plugins_change_detection(f):
+    """This patches the plugins change validation function and makes it behave
+    as if there are always changes in the plugins-to-install lists.
+    """
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        with patch('manager_rest.plugin_update.manager'
+                   '._did_plugins_to_install_change') as _f:
+            _f.return_value = True
+            return f(*args, **kwargs)
+
+    return wrapper
+
+
+@attr(client_min_version=base_test.LATEST_API_VERSION,
+      client_max_version=base_test.LATEST_API_VERSION)
+class PluginsUpdates(BaseServerTestCase):
+    @suppress_plugins_change_detection
+    def test_returns_relevant_plugins_updates(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update1 = self.client.plugins_update.update_plugins(
+            'hello_world')
+        plugins_update2 = self.client.plugins_update.update_plugins(
+            'hello_world', force=True)
+        plugins_update_set = set(p.id for p in self.client.plugins_update
+                                 .list().items)
+        self.assertEqual(2, len(plugins_update_set))
+        self.assertIn(plugins_update1.id, plugins_update_set)
+        self.assertIn(plugins_update2.id, plugins_update_set)
```

Comment:
maybe verify the force also

Reply:
It doesn't save the `force` flag and the mechanism itself is tested in other tests.

Reply:
Now it does, saved and tested